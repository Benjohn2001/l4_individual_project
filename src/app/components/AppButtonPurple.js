import { Component } from "react";
import PropTypes from 'prop-types';
import { TouchableOpacity, Text } from "react-native";

class AppButtonPurple extends Component {
    render(){
        const { title, onPress} = this.props;
		return (
            <TouchableOpacity onPress={onPress} className="bg-darkerPurple h-12 w-10/12 pt-4 rounded-full mx-1.5">
                <Text  className="text-white text-center">{title}</Text>
            </TouchableOpacity>
        );
    }
}

AppButtonPurple.propTypes = {
    title: PropTypes.string.isRequired,
    onPress: PropTypes.func.isRequired
};

export default AppButtonPurple;