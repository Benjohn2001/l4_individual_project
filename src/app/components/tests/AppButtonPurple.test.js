import AppButtonPurple from '../AppButtonPurple';
import { render, screen, fireEvent } from '@testing-library/react-native';
import renderer from "react-test-renderer";

test('Check AppButtonPurple renders correctly and onpress works',()=>{
    const onPressMock = jest.fn();
    render(
        <AppButtonPurple 
            title="press"
            onPress={()=>{
                onPressMock()
            }}
        />,
      );
      fireEvent.press(screen.getByText('press'));
      expect(onPressMock)
      const domTree = renderer.create(
        <AppButtonPurple 
            title="press"
            onPress={()=>{
                onPressMock()
            }}
        />
        ).toJSON();
      expect(domTree).toMatchSnapshot();
})