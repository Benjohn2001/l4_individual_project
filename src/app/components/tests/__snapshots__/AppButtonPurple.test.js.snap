// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check AppButtonPurple renders correctly and onpress works 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#6B4EFF",
      "borderBottomLeftRadius": 9999,
      "borderBottomRightRadius": 9999,
      "borderTopLeftRadius": 9999,
      "borderTopRightRadius": 9999,
      "height": 48,
      "marginLeft": 6,
      "marginRight": 6,
      "opacity": 1,
      "paddingTop": 12,
      "width": "83.333333%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
        },
        Object {
          "textAlign": "center",
        },
      ]
    }
  >
    press
  </Text>
</View>
`;
